src/wallet/wallet_tx_builder.cpp

```diff
@@ -376,11 +376,20 @@ WalletTxBuilder::GetChangeAddress(
                     fvk.GetKnownReceiverTypes());
         },
         [&](const AccountZTXOPattern& acct) -> tl::expected<ChangeAddress, AddressResolutionError> {
-                auto addr = wallet.GenerateChangeAddressForAccount(
+            return wallet.GenerateChangeAddressForAccount(
                     acct.GetAccountId(),
-                    getAllowedChangePools(acct.GetReceiverTypes()));
-            assert(addr.has_value());
-            return addr.value();
+                    getAllowedChangePools(acct.GetReceiverTypes()))
+                .map_error([](auto err) {
+                    switch (err) {
+                        case AccountChangeAddressFailure::DisjointReceivers:
+                            return AddressResolutionError::CouldNotResolveReceiver;
+                        case AccountChangeAddressFailure::TransparentChangeNotPermitted:
+                            return AddressResolutionError::TransparentChangeNotAllowed;
+                        case AccountChangeAddressFailure::NoSuchAccount:
+                            assert(false);
```

Comment:
This should probably throw, or at least `assert(err != AccountChangeAddressFailure::NoSuchAccount)`?

Reply:
I’m fine with either. It’s true, I’m not certain that this case is checked in all cases before we get here.

Just an FYI for the future, I have sellout/zcash#4 (that I don’t think should be in the hotfix), which eliminates some of these failures.